### DISCUSSION TITLE
convert -annotate ignores newline chars since (at least IM 6.9.10

### DISCUSSION THREAD

**Discussion by krauspe** [ImageMagick/ImageMagick#4203] [Help]
OS : CentOS 7.9
IM version  6.9.10

Placing a text string , containing newline chars ("\n")  using -annotate produced on all versions I used before 6.9.10 
multiple lines containing parts of the separated lines
In this version "\n" seams to get ignored !? 

example:
 
convert  -annotate 0 "String1\nString2" infile.jpg outfile.jpg  (no non ascii characters involved)
plots the 2 strings one upon the other , leaving it unreadable

The same command worked fine before with same strings.
I verified this, using another machine running the same command with an older version of IM (6.7.8)

I read the current docs but therer's no change mentioned concerning that behaviour

Regards
PK

**Comment by urban-warrior** [comment]
Try this command:

```
convert -annotate 0x0+10+20 "String1\nString2" infile.jpg outfile.jpg
```

**Comment by windywx** [comment]
I'm running ImageMagick version 6.9.10-68 Q16 x86_64 2020-12-18 on RHEL 7.9, and have the same issue as @krauspe, with "-annotate" ignoring the newline character. I tested the command format suggested by @urban-warrior, but no luck.

`/usr/bin/convert -font Times-Bold -pointsize 48 -fill black -gravity center -annotate 0x0+0+10 "Test String 1\n2 String Test" ./annotate-newline-github-base.png ./annotate-newline-github-out-1.png`

The resulting image still has overlapping text.

![annotate-newline-github-out-1](https://user-images.githubusercontent.com/31943214/109310855-0f386d00-7813-11eb-9972-44451dda2c90.png)

I couldn't find an alternative approach to incorporate the newline, so I just separated the two strings. Not the way I wanted to do the code, but it got the result I needed without losing more time to troubleshooting.

`/usr/bin/convert -font Times-Bold -pointsize 48 -fill black -gravity center -annotate 0x0+0-10 "Test String 1" -annotate 0x0+0+35 "2 String Test" ./annotate-newline-github-base.png ./annotate-newline-github-out-2.png`

![annotate-newline-github-out-2](https://user-images.githubusercontent.com/31943214/109311684-190ea000-7814-11eb-9dea-33e9696d499a.png)

**Comment by fmw42** [comment]
Works fine for me on IM 6.9.12.0 Q16 Mac OSX

`convert -font /Library/fonts/"Times New Roman Bold.ttf" -pointsize 48 -fill black -gravity center -annotate 0x0+0+10 "Test String 1\n 2 String Test" -size 500x500 xc:white x.png`

![x](https://user-images.githubusercontent.com/13458218/109329470-7522f580-780f-11eb-9f6c-6a087ddd5bf4.png)

Note, that this syntax is not correct, though IM 6 is more forgiving than IM 7.  You should read the input right after convert.

Try putting a space after \n

`convert -font /Library/fonts/"Times New Roman Bold.ttf" -pointsize 48 -fill black -gravity center -annotate 0x0+0+10 "Test String 1\n 2 String Test" -size 500x500 xc:white x.png`

Does that help? If not, perhaps your Times font glyphs are corrupt. Try installing another Times family.
